src/Compilers/Shared/BuildServerConnection.cs

```diff
@@ -548,6 +551,26 @@ internal static string GetPipeNameForPathOpt(string compilerExeDirectory)
             return $"{userName}.{(isAdmin ? 'T' : 'F')}.{basePipeName}";
         }
 
+        /// <summary>
+        /// Macs have variable-length temp directories and a small max path
+        /// length for Unix domain sockets.
+        /// </summary>
+        internal static bool IsPipePathTooLong(string pipeName, string tempPath)
```

Comment:
We should also protect VBCSCompiler startup with this check so it gracefully shuts down.